internal/formatters/results/timeline/formatter.go

```diff
@@ -3,57 +3,126 @@
 package timeline
 
 import (
+	"encoding/json"
 	"fmt"
 	"sort"
-	"strings"
 
 	"github.com/cobaltspeech/sdk-cubic/grpc/go-cubic/cubicpb"
+	"github.com/golang/protobuf/ptypes/duration"
 )
 
-// Format generates a list of formatted utterances.
-// The format of each utterance is "start_time_ms|audio_channel_id|1-best transcript".
-//
-// Some assumptions:
-// * The config used in the original transcribe request included, so that timestamps will be available:
-//   &cubicpb.RecognitionConfig{
-//     EnableWordTimeOffsets: true,
-//     EnableRawTranscript: true
-//   }
+// Alternative is a subset of the fields in cubicpb.RecognitionAlternative
+type Alternative struct {
+	StartTime  int64               `json:"start_time"`
+	Confidence float64             `json:"confidence"`
+	Transcript string              `json:"transcript"`
+	Words      []*cubicpb.WordInfo `json:"-"`
+}
+
+// MarshalJSON hides the unfriendly JSON output for duration.Duration
+func (a *Alternative) MarshalJSON() ([]byte, error) {
+	// create aliases so we have all the same fields but none of the methods and don't inherit the original type's MarshalJSON
+	type Alias Alternative
+	type WordAlias cubicpb.WordInfo
+	type W struct {
+		StartTime int64 `json:"start_time"`
+		Duration  int64 `json:"duration"`
+		*WordAlias
+	}
+	out := &struct {
+		Words []*W `json:"words,omitempty"`
+		*Alias
+	}{
+		Alias: (*Alias)(a),
+	}
+
+	for _, word := range a.Words {
+		out.Words = append(out.Words, &W{
+			StartTime: durToMs(word.StartTime),
+			Duration:  durToMs(word.Duration),
+			WordAlias: (*WordAlias)(word),
+		})
+
+	}
+	return json.Marshal(out)
+}
+
+// Result encapsulates the Alternatives for a specific endpointed utterance
+type Result struct {
+	ChannelID int           `json:"channel_id"`
+	Nbest     []Alternative `json:"nbest"`
+}
+
+// ResultFormatter formats a slice of RecognitionResults, e.g. an entire file's worth
+type ResultFormatter interface {
+	Format(results []*cubicpb.RecognitionResult) (string, error)
+}
+
+// Config holds the various options that can be used for this Formatter.
+type Config struct {
+	MaxAlternatives int
+}
+
+// CreateFormatter returns a new Formatter instance with the given configurations
```

Comment:
I'm not in love with this pattern, but it's how we instantiate all the formatters in the formatter repo 
`¯\_(ツ)_/¯`

Reply:
Yeah, fixing that pattern over there is on my todo list.  😅
Why don't you go ahead and do it however you would normally do it here, and we can re-address this later if needed.  Since that pattern is going to be changing, I don't think we need to apply it here.  Especially if they are in separate repos for now.